### DISCUSSION TITLE
How do I use the null object?

### DISCUSSION THREAD

**Discussion by NusryNizam** [friction2d/friction#303] [Q&A] [answered]
I couldn't quite get how the null object can be used. Can anyone shed light with some use cases?

**Comment by pgilfernandez** [comment, marked as answer]
1. Simple tutorial: https://www.youtube.com/watch?app=desktop&v=rGiy7VLPHcY
2. Advanced tutorial: https://www.youtube.com/watch?v=RaM10Vnif6A

I'm also new with Friction and there is no much documentation yet, but, you can search for "Enve tutorials" and find many. Friction is an evolution of Enve you most things are similar and may work in Friction 😉

**Reply by rodlie** [reply]
enve tutorials are currently your only option. Note that the raster paint stuff has been removed and ORA support (compared to enve).

**Reply by NusryNizam** [reply]
Thanks @pgilfernandez @rodlie 

Noted. I'll look up for some enve tutorials
I dabbled in making some basic tutorials for Friction. I hope to make some advanced ones after I learn and experiment more with it.

**Comment by rodlie** [comment]
You can reference null objects from other objects, like this:


https://github.com/user-attachments/assets/632881c9-fd85-4dab-944c-ef8eae6432fb

You can reference the null object using expressions or various path effects.

**Reply by NusryNizam** [reply]
Thanks guys. Didn't know how to link the null object to other objects. Now I know. So many options are hidden 😅
(I just found out the expression editor was already there in the stable version)

**Reply by pgilfernandez** [reply]
About expressions editor: you can also find some Enve tutorials touching it and in this recent discussion you can find some expressions we are starting to use https://github.com/orgs/friction2d/discussions/242

**Reply by NusryNizam** [reply]
Thanks. I'm actively following the discussions here.
This software is a gem.

**Reply by pgilfernandez** [reply]
> This software is a gem.

I do agree

**Reply by pgilfernandez** [reply]
> > Didn't know how to link the null object to other objects
> 
> 
> 
> Perhaps it could be made clearer, I couldn't find it either. I think AE does it directly in the UI by dragging the desired object's property to the null object's property (eg. x position)

I'm planning to create a bunch of short videos showing up simple Friction features but till that moment arrives here is a simple explanation to bind an object (path, group, layer,...) to a null one:

1) Select the object you want to follow the null object:
2) expand it so that you can see the transform properties
3) secondary button on that `transform` and select show "transform effects"
4) you will see now a grey text "transform effect", secondary button here and add the one you want. I suggest you to start with "parent" as it will follow all transforms that the null object does
5) expand the "parent effect" and select the null object to follow
6) done. Now if you move the null object, the patented one will follow it

💪🏻

**Reply by pgilfernandez** [reply]
@rodlie also said this can be done using "expressions". That means you can do it with a script, with all the power and flexibility it offers as you can create custom "parenting", "following", etc. effects with awesome results such as limiting movements in an axis, creating proportional movements, mix properties and do crazy things (bind a rotation value with a X or Y one, use sines/cosines to create bouncing movements,... the possibilities are endless)

**Reply by pgilfernandez** [reply]
a simple video tutorial:

https://github.com/user-attachments/assets/5f45e11b-de83-4b0b-85c0-f41c5d08f85b
